mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/RowMatrix.scala

```diff
@@ -498,6 +498,65 @@ class RowMatrix(
   }
 
   /**
+   * Compute QR decomposition for rowMatrix. The implementation is designed to optimize the QR
+   * decomposition (factorizations) for the RowMatrix of a tall and skinny shape, yet it applies
+   * to RowMatrix in general.
+   *
+   * Reference:
+   *  Austin R. Benson, David F. Gleich, James Demmel. "Direct QR factorizations for tall-and
+   *  -skinny matrices in MapReduce architectures", 2013 IEEE International Conference on Big Data
+   * @param computeQ: whether to computeQ, which is quite expensive.
+   * @return the decomposition result as (Option[Q], R), where Q is a RowMatrix and R is Matrix.
```

Comment:
Similar to SVD, we can have a container for the result. So users can use `Q` and `R` to get the result matrix rather than `_1` and `_2`.